`go test` fails with latest release

**Describe the bug**
Running `go test ./...` fails for me on the latest release `3.0.9`

**To Reproduce**
Steps to reproduce the behavior:
1. Build like this:
```bash
mkdir -p build
export CGO_CPPFLAGS="$CPPFLAGS"
export CGO_CFLAGS="$CFLAGS"
export CGO_CXXFLAGS="$CXXFLAGS"
export CGO_LDFLAGS="$LDFLAGS"
export GOFLAGS="-buildmode=pie -trimpath -ldflags=-linkmode=external -mod=readonly -modcacherw"
go build -o build ./cmd/...
```
2. Run `go test ./...`

**Expected behavior**
All tests pass as in 3.0.8

**Output**
```
go: downloading github.com/stretchr/testify v1.9.0
go: downloading gopkg.in/yaml.v3 v3.0.1
go: downloading github.com/pmezard/go-difflib v1.0.0
ok      github.com/rusq/slackdump/v3    1.220s
ok      github.com/rusq/slackdump/v3/auth       0.006s
ok      github.com/rusq/slackdump/v3/auth/auth_ui       0.005s
ok      github.com/rusq/slackdump/v3/auth/browser       0.005s
ok      github.com/rusq/slackdump/v3/auth/browser/pwcompat      0.003s [no tests to run]
ok      github.com/rusq/slackdump/v3/cmd/slackdump      0.020s
ok      github.com/rusq/slackdump/v3/cmd/slackdump/internal/apiconfig   0.015s
ok      github.com/rusq/slackdump/v3/cmd/slackdump/internal/archive     0.017s
ok      github.com/rusq/slackdump/v3/cmd/slackdump/internal/bootstrap   0.006s
ok      github.com/rusq/slackdump/v3/cmd/slackdump/internal/cfg 0.005s
?       github.com/rusq/slackdump/v3/cmd/slackdump/internal/convertcmd  [no test files]
ok      github.com/rusq/slackdump/v3/cmd/slackdump/internal/diag        0.032s
?       github.com/rusq/slackdump/v3/cmd/slackdump/internal/diag/info   [no test files]
ok      github.com/rusq/slackdump/v3/cmd/slackdump/internal/dump        0.026s [no tests to run]
?       github.com/rusq/slackdump/v3/cmd/slackdump/internal/emoji       [no test files]
ok      github.com/rusq/slackdump/v3/cmd/slackdump/internal/emoji/emojidl       0.012s
?       github.com/rusq/slackdump/v3/cmd/slackdump/internal/export      [no test files]
?       github.com/rusq/slackdump/v3/cmd/slackdump/internal/format      [no test files]
ok      github.com/rusq/slackdump/v3/cmd/slackdump/internal/golang/base 0.022s
?       github.com/rusq/slackdump/v3/cmd/slackdump/internal/golang/help [no test files]
ok      github.com/rusq/slackdump/v3/cmd/slackdump/internal/list        0.027s
?       github.com/rusq/slackdump/v3/cmd/slackdump/internal/man [no test files]
?       github.com/rusq/slackdump/v3/cmd/slackdump/internal/resume      [no test files]
?       github.com/rusq/slackdump/v3/cmd/slackdump/internal/ui  [no test files]
?       github.com/rusq/slackdump/v3/cmd/slackdump/internal/ui/ask      [no test files]
?       github.com/rusq/slackdump/v3/cmd/slackdump/internal/ui/bubbles/btime    [no test files]
?       github.com/rusq/slackdump/v3/cmd/slackdump/internal/ui/bubbles/datepicker       [no test files]
ok      github.com/rusq/slackdump/v3/cmd/slackdump/internal/ui/bubbles/filemgr  0.006s
?       github.com/rusq/slackdump/v3/cmd/slackdump/internal/ui/bubbles/menu     [no test files]
ok      github.com/rusq/slackdump/v3/cmd/slackdump/internal/ui/cfgui    0.021s [no tests to run]
?       github.com/rusq/slackdump/v3/cmd/slackdump/internal/ui/dumpui   [no test files]
ok      github.com/rusq/slackdump/v3/cmd/slackdump/internal/ui/updaters 0.005s
?       github.com/rusq/slackdump/v3/cmd/slackdump/internal/ui/updaters/examples/date   [no test files]
?       github.com/rusq/slackdump/v3/cmd/slackdump/internal/ui/updaters/examples/filenew        [no test files]
?       github.com/rusq/slackdump/v3/cmd/slackdump/internal/ui/updaters/examples/picklist       [no test files]
?       github.com/rusq/slackdump/v3/cmd/slackdump/internal/ui/updaters/examples/string [no test files]
?       github.com/rusq/slackdump/v3/cmd/slackdump/internal/ui/updaters/examples/time   [no test files]
# github.com/rusq/slackdump/v3/internal/network [github.com/rusq/slackdump/v3/internal/network.test]
internal/network/network_test.go:307:51: cannot use func(ctx context.Context) error {…} (value of type func(ctx context.Context) error) as func() error value in argument to WithRetry
ok      github.com/rusq/slackdump/v3/cmd/slackdump/internal/view        0.020s
?       github.com/rusq/slackdump/v3/cmd/slackdump/internal/wizard      [no test files]
ok      github.com/rusq/slackdump/v3/cmd/slackdump/internal/workspace   0.020s
?       github.com/rusq/slackdump/v3/cmd/slackdump/internal/workspace/workspaceui       [no test files]
ok      github.com/rusq/slackdump/v3/downloader 0.031s
?       github.com/rusq/slackdump/v3/export     [no test files]
ok      github.com/rusq/slackdump/v3/internal/cache     0.076s
ok      github.com/rusq/slackdump/v3/internal/chunk     0.012s
ok      github.com/rusq/slackdump/v3/internal/chunk/chunktest   0.007s
?       github.com/rusq/slackdump/v3/internal/chunk/control     [no test files]
ok      github.com/rusq/slackdump/v3/internal/chunk/dirproc     0.007s
ok      github.com/rusq/slackdump/v3/internal/chunk/obfuscate   0.057s
ok      github.com/rusq/slackdump/v3/internal/chunk/state       0.004s
ok      github.com/rusq/slackdump/v3/internal/chunk/transform   0.005s
ok      github.com/rusq/slackdump/v3/internal/chunk/transform/fileproc  0.006s
ok      github.com/rusq/slackdump/v3/internal/convert   0.005s
ok      github.com/rusq/slackdump/v3/internal/edge      0.010s
ok      github.com/rusq/slackdump/v3/internal/fasttime  0.003s
?       github.com/rusq/slackdump/v3/internal/fixtures  [no test files]
?       github.com/rusq/slackdump/v3/internal/fixtures/fixchunks        [no test files]
?       github.com/rusq/slackdump/v3/internal/fixtures/fixgen   [no test files]
ok      github.com/rusq/slackdump/v3/internal/format    0.004s
?       github.com/rusq/slackdump/v3/internal/mocks/mock_auth   [no test files]
?       github.com/rusq/slackdump/v3/internal/mocks/mock_cache  [no test files]
?       github.com/rusq/slackdump/v3/internal/mocks/mock_io     [no test files]
?       github.com/rusq/slackdump/v3/internal/mocks/mock_os     [no test files]
ok      github.com/rusq/slackdump/v3/internal/nametmpl  0.003s
FAIL    github.com/rusq/slackdump/v3/internal/network [build failed]
ok      github.com/rusq/slackdump/v3/internal/osext     0.002s
ok      github.com/rusq/slackdump/v3/internal/primitive 0.002s
ok      github.com/rusq/slackdump/v3/internal/source    0.005s
ok      github.com/rusq/slackdump/v3/internal/structures        0.005s
ok      github.com/rusq/slackdump/v3/internal/structures/files  0.002s
ok      github.com/rusq/slackdump/v3/internal/viewer    0.003s
ok      github.com/rusq/slackdump/v3/internal/viewer/renderer   0.004s
?       github.com/rusq/slackdump/v3/internal/viewer/renderer/functions [no test files]
?       github.com/rusq/slackdump/v3/mocks/mock_downloader      [no test files]
?       github.com/rusq/slackdump/v3/mocks/mock_processor       [no test files]
?       github.com/rusq/slackdump/v3/processor  [no test files]
ok      github.com/rusq/slackdump/v3/stream     0.156s
?       github.com/rusq/slackdump/v3/stream/mock_stream [no test files]
ok      github.com/rusq/slackdump/v3/types      0.002s
FAIL
```

**Desktop (please complete the following information):**
 - OS: Arch Linux

**Additional context**
I notice the issue when bumping the release version for [the PKGBUILD in AUR](https://aur.archlinux.org/packages/slackdump). Everything is build on a clean chroot env.

--- a/.github/workflows/docker.yml
+++ b/.github/workflows/docker.yml
@@ -5,7 +5,7 @@ on:
         types:
             - published
     push:
-        branches: [master]
+        branches: [master,v3.0.x]
 
 
 # reference:

--- a/.github/workflows/docker.yml
+++ b/.github/workflows/docker.yml
@@ -5,7 +5,7 @@ on:
         types:
             - published
     push:
-        branches: [master,v3.0.x]
+        branches: [master, v3.0.x]
 
 
 # reference:

--- a/.github/workflows/go.yml
+++ b/.github/workflows/go.yml
@@ -2,9 +2,9 @@ name: Go
 
 on:
   push:
-    branches: [ master ]
+    branches: [ master, v3.0.x ]
   pull_request:
-    branches: [ master ]
+    branches: [ master, v3.0.x ]
 
 jobs:
 

--- a/.github/workflows/shellcheck.yml
+++ b/.github/workflows/shellcheck.yml
@@ -4,9 +4,9 @@ name: Shellcheck
 
 on:
   push:
-    branches: [master]
+    branches: [master, v3.0.x]
   pull_request:
-    branches: [master]
+    branches: [master, v3.0.x]
 
 permissions:
   contents: read

--- a/internal/network/network_test.go
+++ b/internal/network/network_test.go
@@ -304,7 +304,7 @@ func TestWithRetry(t *testing.T) {
 		var retries int
 
 		ctx := context.Background()
-		err := WithRetry(ctx, rate.NewLimiter(1, 1), 3, func(ctx context.Context) error {
+		err := WithRetry(ctx, rate.NewLimiter(1, 1), 3, func() error {
 			err := reterr[retries]
 			if err != nil {
 				retries++

--- a/internal/network/network_test.go
+++ b/internal/network/network_test.go
@@ -303,6 +303,8 @@ func TestWithRetry(t *testing.T) {
 		reterr := []error{io.ErrUnexpectedEOF, io.ErrUnexpectedEOF, nil}
 		var retries int
 
+		// dummy line to trigger the build.
+
 		ctx := context.Background()
 		err := WithRetry(ctx, rate.NewLimiter(1, 1), 3, func() error {
 			err := reterr[retries]

--- a/internal/network/network_test.go
+++ b/internal/network/network_test.go
@@ -302,9 +302,6 @@ func TestWithRetry(t *testing.T) {
 
 		reterr := []error{io.ErrUnexpectedEOF, io.ErrUnexpectedEOF, nil}
 		var retries int
-
-		// dummy line to trigger the build.
-
 		ctx := context.Background()
 		err := WithRetry(ctx, rate.NewLimiter(1, 1), 3, func() error {
 			err := reterr[retries]

